### ISSUE TITLE
Syntastic suddenly stopped working when I removed my node_modules folder

### ISSUE THREAD

**Issue by stavenko** [vim-syntastic/syntastic#2042]
Hello. 
My syntastic doesn't work anymore.
Here's some details, that I've collected:
Vim:
```
mvim --version
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Jun 22 2017 07:10:35)
MacOS X (unix) version
Included patches: 1-648
Compiled by Homebrew
Huge version with MacVim GUI.  Features included (+) or not (-):
+acl             +find_in_path    -mouse_sysmouse  -tag_any_white
+arabic          +float           +mouse_urxvt     +tcl
+autocmd         +folding         +mouse_xterm     +termguicolors
+balloon_eval    -footer          +multi_byte      +terminfo
+browse          +fork()          +multi_lang      +termresponse
++builtin_terms  +fullscreen      -mzscheme        +textobjects
+byte_offset     -gettext         +netbeans_intg   +timers
+channel         -hangul_input    +num64           +title
+cindent         +iconv           +odbeditor       +toolbar
+clientserver    +insert_expand   +packages        +transparency
+clipboard       +job             +path_extra      +user_commands
+cmdline_compl   +jumplist        +perl            +vertsplit
+cmdline_hist    +keymap          +persistent_undo +virtualedit
+cmdline_info    +lambda          +postscript      +visual
+comments        +langmap         +printer         +visualextra
+conceal         +libcall         +profile         +viminfo
+cryptv          +linebreak       +python          +vreplace
+cscope          +lispindent      -python3         +wildignore
+cursorbind      +listcmds        +quickfix        +wildmenu
+cursorshape     +localmap        +reltime         +windows
+dialog_con_gui  -lua             +rightleft       +writebackup
+diff            +menu            +ruby            -X11
+digraphs        +mksession       +scrollbind      -xfontset
+dnd             +modify_fname    +signs           +xim
-ebcdic          +mouse           +smartindent     -xpm
+emacs_tags      +mouseshape      +startuptime     -xsmp
+eval            +mouse_dec       +statusline      -xterm_clipboard
+ex_extra        -mouse_gpm       -sun_workshop    -xterm_save
+extra_search    -mouse_jsbterm   +syntax          
+farsi           +mouse_netterm   +tag_binary      
+file_in_path    +mouse_sgr       +tag_old_static  

```
~/.vimrc
```
set statusline+=%#warningmsg#
set statusline+=%{SyntasticStatuslineFlag()}
set statusline+=%*

let g:indent_guides_auto_colors=0
let g:solarized_termcolors=255
let g:syntastic_debug=33
let g:syntastic_check_on_open=1
let g:syntastic_always_populate_loc_list = 1
let g:syntastic_javascript_checkers=['eslint']
```

Here's what sytastic says when I open the file:

```
syntastic: 28.712309: g:syntastic_version = '3.8.0-63 (Vim 800, Darwin, GUI)'
syntastic: 28.712433: &shell = '/bin/bash', &shellcmdflag = '-c', &shellpipe = '2>&1| tee', &shellquote = '', &shellredir = '>%s 2>&1', &shelltemp = 1, &shellxquote = '', &autochdir = 0, &shellxescape = ''
syntastic: 28.712769: UpdateErrors (auto): default checkers
syntastic: 28.713031: CacheErrors: default checkers
syntastic: 28.713550: g:syntastic_aggregate_errors = 0
syntastic: 28.713678: $TERM = ''
syntastic: 28.713749: $TMPDIR = '/var/folders/fm/bfwjc7p15m3cnyxcxxyshnv5tjdvk5/T/'
syntastic: 28.713819: $PATH = '/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin'
syntastic: 28.713898: getcwd() = '/Users/vstavenko/projects'
syntastic: 28.978440: system: command run in 0.260102s
syntastic: 28.978872: javascript/eslint: getVersion: 'eslint --version': ['v4.2.0', '']
syntastic: 28.979180: javascript/eslint: eslint version = [4, 2, 0]
syntastic: 28.979527: CacheErrors: Invoking checker: javascript/eslint
syntastic: 28.980608: SyntasticMake: called with options: {'postprocess': ['guards'], 'errorformat': '%E%f: line %l\, col %c\, Error - %m,%W%f: line %l\, col %c\, Warning - %m', 'makeprg': 'eslint -f compact WebCC/ClinCheck/Material.js'}
syntastic: 29.256815: system: command run in 0.275734s
syntastic: 29.258153: getLocList: checker javascript/eslint returned 1
syntastic: 29.258424: getLocList: checker javascript/eslint run in 0.278735s   
```

Here's how my vim sees eslint:

```
:!which eslint
/usr/local/bin/eslint 
```

And this is how eslint works from my vim:

```
:!eslint -f compact ./projects/WebCC/ClinCheck/Material.js 

Oops! Something went wrong! :(

ESLint couldn't find the plugin "eslint-plugin-react". This can happen for a couple different reasons:

1. If ESLint is installed globally, then make sure eslint-plugin-react is also installed globally. A globally-installed ESLint cannot find a locally-installed plugin.

2. If ESLint is installed locally, then it's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm i eslint-plugin-react@latest --save-dev

If you still can't figure out the problem, please stop by https://gitter.im/eslint/eslint to chat with the team.
```

Oh, wait... This seems like I found a problem, whilst I was collecting data for correct questions. 

Sorry guys for bothering. Maybe others will notice this as a way to discover errors. 
Syntastic is the best!

**Comment by lcd047** [comment]
TL;DR: syntastic needs third-party linters to be in a working state.
